### PR TITLE
Expose noobaa endpoint resource management to OCS operator

### LINKED ISSUE #403: Support override noobaa-endpoint resources in StorageCluster.spec.resources map
NooBaa endpoint pods have a spec to override the default resources in the noobaa CR spec:
https://github.com/noobaa/noobaa-operator/blob/8fb75596137a9359e4c00cc7c6f84eb3c5e5d8a9/pkg/apis/noobaa/v1alpha1/noobaa_types.go#L153

Need to add this to the StorageCluster.spec.resources map and reconcile it to the NooBaa CR - 
https://github.com/openshift/ocs-operator/blob/b3bf10e90e5cb14f5509c9c049f1eeba0e16855b/pkg/controller/storagecluster/noobaa_system_reconciler.go#L103-L105

CC @nimrod-becker @nb-ohad

### PR DESCRIPTION
**Pull request by nb-ohad**
A NooBaa CR can describe memory and CPU resource management (requests/limits) for deployed NooBaa endpoints.

This PR exposes these options in the StorageCluster CR and add reconcile code to update the NooBaa CR accordingly  

Fixes #403

### CODE REVIEW

File: pkg/controller/defaults/resources.go [hunk 1]
```diff
@@ -79,10 +79,20 @@ var (
 				corev1.ResourceMemory: resource.MustParse("4Gi"),
 			},
 		},
-		"noobaa-db-vol": corev1.ResourceRequirements{
+		"noobaa-db-vol": {
 			Requests: corev1.ResourceList{
 				corev1.ResourceStorage: resource.MustParse("50Gi"),
 			},
 		},
+		"noobaa-endpoint": {
```
Comment (guymguym) [inline pkg/controller/defaults/resources.go]: @nb-ohad should we re-adjust the noobaa-core resources now that endpoints have split from it?

Comment (nb-ohad) [reply pkg/controller/defaults/resources.go]: @guymguym I was thinking this is the readjusted resources, Please correct me If I am wrong.

Comment (guymguym) [reply pkg/controller/defaults/resources.go]: Better check it

**Comment by guymguym** [comment]
@nb-ohad @nimrod-becker bump - what is missing here?

File: pkg/controller/defaults/resources.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ var (
 	// DaemonResources map contains the default resource requirements for the
 	// various OCS daemons
 	DaemonResources = map[string]corev1.ResourceRequirements{
-		"osd": corev1.ResourceRequirements{
```
Comment (jarrpa) [inline pkg/controller/defaults/resources.go:12]: Why are we making these changes?

File: pkg/controller/defaults/resources.go [hunk 3]
```diff
@@ -79,10 +79,20 @@ var (
...
+		"noobaa-endpoint": {
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("2Gi"),
+			},
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("2Gi"),
+			},
+		},
```
Comment (jarrpa) [inline pkg/controller/defaults/resources.go:96]: If the above changes are valid, move this to a separate commit.

Comment (nb-ohad) [reply pkg/controller/defaults/resources.go:96]: It is in a separate commit, Do you mean a separate PR?

File: pkg/controller/defaults/resources.go (hunk 2 above)
Comment (nb-ohad) [reply pkg/controller/defaults/resources.go:12]: Its a change in `gofmt` that remove unnecessary type declarations

Comment (jarrpa) [reply pkg/controller/defaults/resources.go:12]: Ah, okay.

File: pkg/controller/defaults/resources.go (hunk 3 above)
Comment (jarrpa) [reply pkg/controller/defaults/resources.go:96]: The issue is that the gofmt changes are unrelated to the NooBaa endpoint changes. The gofmt changes should go into their own commit, with a message explaining why they're happening.
